### ISSUE TITLE
1.6-rc.7 update -  MongoError: authentication fail

### ISSUE THREAD

**Issue by benlavalley** [meteor/meteor#9219]
Updated from 1.6-rc.6 to 1.6-rc.7, got a authentication error in my production app when trying to use a remote Mongo DB server at 3.2.17 (Ubuntu 14 LTS). Not sure if I missed a memo about my mongo version dropping from support or if this is a legitimate issue? 

Going to be working on a repro now but wanted to get it posted considering there were some changes I saw with mongo from rc.6->rc.7, particularly relating to a fix for [oplog tailing](https://github.com/mongodb-js/mongodb-core/pull/224), and my authentication issue appears to only be with the oplogger.

App on my dev box works fine with the default local mongodb w/localhost auth. When I run it with a script to set the DB connection string to my production server, I get the same mongo auth error (OS X 10.2.6). 

My production box downgraded back to 1.6-rc.6 and is authenticating fine again.

Connection strings look like:

> export MONGO_URL="mongodb://asdfl:asdf@asdf.com:27020/admin?autoReconnect=true&replicaSet=nibbldbrs0&readPreference=primaryPreferred&w=majority"
> export MONGO_OPLOG_URL="mongodb://oplogger:asdf@asdf.com:27020/local?authSource=admin&replicaSet=nibbldbrs0"

Auth error in mongo logs:
`
2017-10-12T16:07:26.034-0400 I ACCESS   [conn5849] SCRAM-SHA-1 authentication failed for oplogger on admin from client 123.123.123.123 ; AuthenticationFailed: SCRAM-SHA-1 authentication failed, storedKey mismatch`

Meteor error:
```
W20171012-16:13:24.650(-4)? (STDERR) MongoError: authentication fail
W20171012-16:13:24.650(-4)? (STDERR)     at Function.MongoError.create (/Users/ben/.meteor/packages/npm-mongo/.2.2.32.22x45z.ribzlj++os+web.browser+web.cordova/npm/node_modules/mongodb-core/lib/error.js:31:11)
W20171012-16:13:24.650(-4)? (STDERR)     at /Users/ben/.meteor/packages/npm-mongo/.2.2.32.22x45z.ribzlj++os+web.browser+web.cordova/npm/node_modules/mongodb-core/lib/topologies/replset.js:1226:38
W20171012-16:13:24.650(-4)? (STDERR)     at /Users/ben/.meteor/packages/npm-mongo/.2.2.32.22x45z.ribzlj++os+web.browser+web.cordova/npm/node_modules/mongodb-core/lib/connection/pool.js:760:7
W20171012-16:13:24.651(-4)? (STDERR)     at /Users/ben/.meteor/packages/npm-mongo/.2.2.32.22x45z.ribzlj++os+web.browser+web.cordova/npm/node_modules/mongodb-core/lib/connection/pool.js:736:20
W20171012-16:13:24.651(-4)? (STDERR)     at finish (/Users/ben/.meteor/packages/npm-mongo/.2.2.32.22x45z.ribzlj++os+web.browser+web.cordova/npm/node_modules/mongodb-core/lib/auth/scram.js:160:16)
W20171012-16:13:24.651(-4)? (STDERR)     at handleEnd (/Users/ben/.meteor/packages/npm-mongo/.2.2.32.22x45z.ribzlj++os+web.browser+web.cordova/npm/node_modules/mongodb-core/lib/auth/scram.js:170:7)
W20171012-16:13:24.651(-4)? (STDERR)     at /Users/ben/.meteor/packages/npm-mongo/.2.2.32.22x45z.ribzlj++os+web.browser+web.cordova/npm/node_modules/mongodb-core/lib/auth/scram.js:261:11
W20171012-16:13:24.651(-4)? (STDERR)     at /Users/ben/.meteor/packages/npm-mongo/.2.2.32.22x45z.ribzlj++os+web.browser+web.cordova/npm/node_modules/mongodb-core/lib/connection/pool.js:469:18
W20171012-16:13:24.651(-4)? (STDERR)     at _combinedTickCallback (internal/process/next_tick.js:131:7)
W20171012-16:13:24.652(-4)? (STDERR)     at process._tickCallback (internal/process/next_tick.js:180:9)
```

**Comment by benjamn** [comment]
Related: https://github.com/meteor/meteor/pull/9200 (any thoughts, @glasser?)

**Comment by benlavalley** [comment]
Updated my production DB to 3.4.9, didn't seem to help. I also tried running changeUserPassword in Mongo (figured maybe it needed to update it's stored hashes or something)... no dice.

**Comment by glasser** [comment]
@benlavalley can you tell me what versions are reported for `npm-mongo` in `.meteor/versions` both when it's working and failing?

Note that a 1.5.2.2 project will pull in the new npm-mongo so that's not the most surprising thing in the world. I believe 2.2.30 was the last version before my fix and 2.2.31 and 2.2.32 are the ones with my fix. Can you test against each of these separately and tell me what you discover?  (eg, add `npm-mongo@=2.2.30` to your `.meteor/packages`.

**Comment by benlavalley** [comment]
Was about to ask how to downgrade that mongo package.

Yeah that resolves it. Good and bad news I guess :)

**Comment by abernix** [comment]
It looks like the fix from `mongodb-core` is on it's way already: https://github.com/mongodb-js/mongodb-core/commit/c532c95b7755bfeaf0dcb44768edc146192f242a

**Comment by benlavalley** [comment]
> 2.1.17 2017-10-11
> -----------------
> * fix a typo that completely broke SCRAM-SHA1 authentication

BRUTAL.

**Comment by benjamn** [comment]
Wow, ok!

**Comment by glasser** [comment]
Eek!

You folks on getting this one out to Meteor?

**Comment by glasser** [comment]
BTW I'm not sure if this was in 2.2.31 (ie my form of mongodb-core) or not

**Comment by glasser** [comment]
Looks like no, so only 2.2.32 is bad.

**Comment by benlavalley** [comment]
Looks like 2.2.33 was just posted, or at least my server just pulled it down running a meteor update.  Working fine.

**Comment by benjamn** [comment]
Just published `npm-mongo@2.2.33` with `mongodb@2.2.33` and `mongodb-core@2.1.17`: 525cbc33b32acb1d8e7c7571320a57376e712d32
